tools/data-validator/src/main/java/data/validator/DataValidator.java

```diff
@@ -30,50 +30,60 @@
 
 
     private static final String SELECT_COLUMN_NAMES = "SELECT " + COLUMN_NAME_SELECTOR + ", " + COLUMN_INDEX_TYPE + " FROM " + SYSTEM_SCHEMA_COLUMNS_TABLE +
-            "  WHERE keyspace_name =  ?  AND columnfamily_name = ? ;";
+            "  WHERE " + KEYSPACE_NAME_LABEL + "=  ?  AND " + TABLE_NAME_LABEL + "= ? ;";
 
 
     public void validate(String sourceTableName, String targetTableName, int threadsCount) {
         Session session = null;
         Session targetSession = null;
         ExecutorService executorService = null;
         try {
+            int progressCounter = 0;
             executorService = Executors.newFixedThreadPool(threadsCount);
             session = sourceCassandraConnectionProvider.getSession();
             List<String> primaryKeys = getPrimaryKeysNames(sourceTableName, session);
-            String selectPrimaryKeysFromSourceTable = CQLBuilder.constructSelectPrimaryKeysFromSourceTable(sourceTableName, primaryKeys);
-            PreparedStatement sourceSelectStatement = session.prepare(selectPrimaryKeysFromSourceTable);
-            sourceSelectStatement.setConsistencyLevel(sourceCassandraConnectionProvider.getConsistencyLevel());
-            BoundStatement boundStatement = sourceSelectStatement.bind();
-            ResultSet rs = session.execute(boundStatement);
+            ResultSet rs = getPrimaryKeysFromSourceTable(sourceTableName, session, primaryKeys);
             Iterator<Row> iterator = rs.iterator();
 
-            String matchCountStatementCQL = CQLBuilder.getMatchCountStatementFromTargetTable(targetTableName, primaryKeys);
             targetSession = targetCassandraConnectionProvider.getSession();
-            PreparedStatement matchCountStatementCQLStatement = targetSession.prepare(matchCountStatementCQL);
-            matchCountStatementCQLStatement.setConsistencyLevel(targetCassandraConnectionProvider.getConsistencyLevel());
-            BoundStatement matchingBoundStatement = matchCountStatementCQLStatement.bind();
+            BoundStatement matchingBoundStatement = prepareBoundStatementForMatchingTargetTable(targetTableName, targetSession, primaryKeys);
 
             while (iterator.hasNext()) {
                 Row row = iterator.next();
                 Callable<Long> callable = new RowValidatorJob(targetSession, primaryKeys, matchingBoundStatement, row);
                 Future<Long> future = executorService.submit(callable);
-                System.out.println("Thread tarek " + future.get());
+                future.get();
+                progressCounter++;
+                if (progressCounter % PROGRESS_COUNTER == 0)
+                    System.out.println("The data is matching properly till now and the progress will continue for source table " + sourceTableName + " and target table " + targetTableName + " ....");
             }
```

Comment:
When running for a bigger table the same message is repeated every PROGRESS_COUNTER rows but it does not tell anything about the progress, anything about the number of already checked rows. Such information should be usefull because you more or less know where you are even if you don't know the exact number of rows to be chceked.